waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -202,6 +216,43 @@ public DatabaseMapping databaseMapping(@NotNull String databaseName) throws NoSu
         "Waggle Dance error no database mapping available tried to map database '" + databaseName + "'");
   }
 
+  @Override
+  public void checkTable(String databaseName, String tableName, DatabaseMapping mapping)
+      throws NoSuchObjectException {
+    String databasePrefix = mapping.getDatabasePrefix();
+    String transformedDbName = mapping.transformInboundDatabaseName(databaseName);
+    if (!isTableWhitelisted(databasePrefix, transformedDbName, tableName)) {
+      throw new NoSuchObjectException(String.format("%s.%s table not found", databaseName, tableName));
+    }
+  }
+
+  @Override
+  public List<String> filterTables(String databaseName, List<String> tableNames, DatabaseMapping mapping) {
+    List<String> filteredTables = new ArrayList<>();
+    String databasePrefix = mapping.getDatabasePrefix();
+    String transformedDb = mapping.transformInboundDatabaseName(databaseName);
+    for (String table: tableNames) {
+      if (isTableWhitelisted(databasePrefix, transformedDb, table)) {
+        filteredTables.add(table);
```

Comment:
See my comment on the javadoc on the interface above, I'm a bit confused but maybe it's because I haven't had my morning coffee yet ;) This looks like it's adding a table that is whitelisted to something that needs to be filtered out but I think the intention is actually the opposite and this is returning tables that are "vaild"? Perhaps there's some confusion (or I'm misunderstanding) what "filter" means in this context.

Reply:
I guess I could rename `filteredTables` to `allowedTables`?

Reply:
Yeah, if I understand it correctly, what it's doing is `filterOutUnmappedTables` but that doesn't read very nicely ;) Let's try your suggestion and see what that looks like along with the javadoc.